File: container-with-most-water/container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int maxarea =0,l=0,r=height.Length-1;
        while(l<r)
        {
            maxarea = Math.Max(maxarea,Math.Min(height[l],height[r]) * (r-l));
            if(height[l] < height[r])
            {
                l++;
            }
            else
            {
                r--;
            }
        }
        return maxarea;
        
    }
}

File: twopointers1/src/containerwithmostwater.java
class Solution {
    public int maxArea(int[] height) {
        if(height == null || height.length== 0){
            return 0;
        }
        int max=0;
        int left=0;
        int right=height.length-1;

        while(left<right){
            max= Math.max(max,(right-left)*Math.min(height[left],height[right]));
            if(height[left]<height[right]){
                left+=1;
            }
            else{
                right-=1;
            }
        }
        return max;
    }
}

//time complexity=o(n)
//space complexity=o(1)